### ISSUE TITLE
Blocked Accounts Being Refollowed

### ISSUE THREAD

**Issue by Zen00** [raulrene/Twitter-ContestJS-bot#3]
So I built an extensive block list in order to help cut down on the amount of spam I retweeted, however there seems to be a problem with the way the script handles retweeting blocked accounts.

http://snag.gy/jc1dF.jpg

In this screenshot you can see me retweeting speed_racer420, however I had previously blocked him in order to stop from tweeting his material. The script re-added him to my follow list despite this. Any idea why the block function isn't working?

**Comment by raulrene** [comment]
Hi @Zen00 ,

Can you please check again that this is your case? Maybe block a user that you see has recently retweeted something that the bot is likely to retweet, and see if the bot retweets it. I double checked now, by manually blocking some users and then checking with the bot, and it did not retweet any of those tweets. They are filtered out in the search function.

**Comment by Zen00** [comment]
I'll have to try sometime later. My app got blocked because of too much spam or something and I have to get it unblocked. ;P

Anyways, to specify some accounts that keep popping back up, speed_racer420 and sswifttgamingg.

**Comment by henhouse** [comment]
Hi @Zen00, I fixed this sort of in 5755ca1. The bot won't retweet tons of garbage anymore like you posted on your screenshot based on some threshold parameters I added, which are configurable.

Additionally, the bot can be set to auto-block other bots it detects they are bots and that takes effect immediately by adding them to the block list array. The bot currently isn't programmed to reload your block list from Twitter in the event you block people through the website, but if this is a change you think is needed, I could have the bot periodically reload the block list from Twitter to ensure that it's up-to-date while running.
